queries/javascript.core.scm

```diff
@@ -481,24 +480,33 @@
 ;;!  ^^^^^
 ;;!! new Foo()
 ;;!  ^^^^^^^^^
-[
-  (call_expression)
-  (new_expression)
-] @functionCall
+(
+  [
+    (call_expression)
+    (new_expression)
+  ] @functionCall @_.domain.start
+  ";"? @_.domain.end
```

Comment:
Semi was missing from domain

Reply:
Oh, right this was the bugs I pointed out in my earlier comments 😅. Why would semi be part of the domain? Semi is part of the containing statement; a call is an expression

Reply:
I did some testing on this, and here's the behaviour this PR introduces

<img width="274" alt="image" src="https://github.com/cursorless-dev/cursorless/assets/755842/10de9da0-462e-4e9b-810c-33f534e2d4fd">

Note how it only extends the domain to include the semicolon for

- the outermost call, and
- only if the entire statement is a function call.

In particular, when it's the value of an assignment, or part of a bigger expression, it doesn't include the semicolon

There is a certain logic to that, but might be surprising? Also might be surprising if you say "previous call" and that doesn't work? I don't have a strong feeling one way or another. I also might consider going a step further and including the semicolon on the content range itself

If we do keep this behaviour, let's capture the semantics from that screenshot in a scope fixture. I tried putting all four of those lines in a scope fixture, and man we really aren't optimized for more than one scope per fixture 😅. I wonder if we should

1. Only keep the empty marker lines if they're inside a scope, or
2. Not bother to show lines outside of the scope; that's kinda what the line numbers are for

cc/ @josharian on both topics mentioned here 😊

Reply:
For now I would be fine with to just reverting the domains so we can actually merge this. Changing the domain of call seems totally separate from this pull request.

Dropping lines outside of the scope is totally be fine for me

I still haven't heard of any way to actually determine token removal. And I I'm not sure if we not losing information if we do that.

Reply:
Call domain reverted

Reply:
> I would be supportive of that as a first step. @josharian any objections? If not, I'll file an issue to trackmr

Seems worth trying. If it creates other issues, we can always adjust from there.

Reply:
Filed

- https://github.com/cursorless-dev/cursorless/issues/2139
- https://github.com/cursorless-dev/cursorless/issues/2140